packages/core/components/RichTextEditor/lib/build-editor-state.ts

```diff
@@ -0,0 +1,18 @@
+import { Editor } from "@tiptap/react";
+import { RichTextConfigType } from "../../../types";
+
+export function buildEditorState(
+  editor: Editor,
+  menuConfig: RichTextConfigType["menu"]
+) {
+  const state: Record<string, boolean> = {};
+
+  Object.values(menuConfig)
+    .flat()
+    .forEach((item: any) => {
+      if (item.state) state[item.title] = item.state(editor) ?? false;
+      if (item.can) state[item.title + "Can"] = item.can(editor) ?? false;
```

Comment:
I find this "Can" state stuff a bit confusing. Is this a fair comparison? `.state()` is active state, and `.can()` is whether the control is currently enabled?

Reply:
State:  whether or not the selected text is active e.g. bold is selected
Can: whether or not the action can be performed by the loaded extensions